satpy/readers/__init__.py

```diff
@@ -709,6 +709,9 @@ def load_readers(filenames=None, reader=None, reader_kwargs=None,
     """
     reader_instances = {}
     reader_kwargs = reader_kwargs or {}
+    reader_kwargs_without_filter = reader_kwargs.copy()
+    reader_kwargs_without_filter.pop('filter_parameters', None)
```

Comment:
You could do:

```
filter_parameters = reader_kwargs.pop('filter_parameters', None)
```

and pass reader_kwargs and filter_parameters to the reader creation, but only reader_kwargs to the file handlers. Saves a copy, but not a big deal.

Reply:
ok, so, we need to decide, what are we rooting for ? `filter_parameters` as standalone parameters or part of `reader_kwargs`?

Reply:
I was just making a suggestion to clean up the code. It is the same interface on the reader and to this `load_readers` function. I'm saying you don't need to copy the dictionary if you pass filter_parameters explicitly:

```
filter_parameters = reader_kwargs.pop('filter_parameters', None)
reader = reader_cls(filter_parameters=filter_parameters, **reader_kwargs)
...
reader.create_filehandlers(**reader_kwargs)
```

Reply:
yes, agreed, but I'd like to take this opportunity to clarify things :) what is the "right" way ? This way we could maybe add a deprecation warning or something ?

Reply:
I think in the long-long run we remove the ability to filter things through `load_readers` (and through that the `Scene.__init__`.

In the short term the user should be specifying `filter_parameters` as a separate kwarg but the `Scene.__init__` already has a deprecation warning for this and moves `start_time/end_time/area` to `filter_parameters` for the user. If you want to move `filter_parameters` outside of `reader_kwargs` and make it a separate/explicit kwarg I guess I'd be ok with that and maybe would prefer that.